OpenRA.Mods.Common/Traits/Render/WithAttackAnimation.cs

```diff
@@ -53,12 +54,20 @@ public WithAttackAnimation(ActorInitializer init, WithAttackAnimationInfo info)
 			armament = init.Self.TraitsImplementing<Armament>()
 				.Single(a => a.Info.Name == info.Armament);
 			wsb = init.Self.Trait<WithSpriteBody>();
+
+			if (init.Self.TraitOrDefault<AttackTurreted>() != null)
```

Comment:
>  I wonder if we want to ensure in some way that we use the correct turret though

Yes, we definitely do.  Otherwise, this would not work on multi-turreted units.
I've suggested my preferred approach for this in my comment just above yours.